// @azure/cognitiveservices-customvision-training/dist/cognitiveservices-customvision-training.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@azure/ms-rest-js')) :
    typeof define === 'function' && define.amd ? define(['exports', '@azure/ms-rest-js'], factory) :
    (global = global || self, factory((global.Azure = global.Azure || {}, global.Azure.CognitiveservicesCustomvisionTraining = {}), global.msRest));
}(this, function (exports, msRest) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    /*
     * Copyright (c) Microsoft Corporation. All rights reserved.
     * Licensed under the MIT License. See License.txt in the project root for license information.
     *
     * Code generated by Microsoft (R) AutoRest Code Generator.
     * Changes may cause incorrect behavior and will be lost if the code is regenerated.
     */

    var index = /*#__PURE__*/Object.freeze({

    });

    /*
     * Copyright (c) Microsoft Corporation. All rights reserved.
     * Licensed under the MIT License. See License.txt in the project root for license information.
     *
     * Code generated by Microsoft (R) AutoRest Code Generator.
     * Changes may cause incorrect behavior and will be lost if the code is regenerated.
     */
    var Domain = {
        serializedName: "Domain",
        type: {
            name: "Composite",
            className: "Domain",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                name: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                type: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "type",
                    type: {
                        name: "String"
                    }
                },
                exportable: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "exportable",
                    type: {
                        name: "Boolean"
                    }
                },
                enabled: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "enabled",
                    type: {
                        name: "Boolean"
                    }
                }
            }
        }
    };
    var ImageTagCreateEntry = {
        serializedName: "ImageTagCreateEntry",
        type: {
            name: "Composite",
            className: "ImageTagCreateEntry",
            modelProperties: {
                imageId: {
                    nullable: false,
                    serializedName: "imageId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagId: {
                    nullable: false,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                }
            }
        }
    };
    var ImageTagCreateBatch = {
        serializedName: "ImageTagCreateBatch",
        type: {
            name: "Composite",
            className: "ImageTagCreateBatch",
            modelProperties: {
                tags: {
                    serializedName: "tags",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTagCreateEntry"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageTagCreateSummary = {
        serializedName: "ImageTagCreateSummary",
        type: {
            name: "Composite",
            className: "ImageTagCreateSummary",
            modelProperties: {
                created: {
                    nullable: true,
                    serializedName: "created",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTagCreateEntry"
                            }
                        }
                    }
                },
                duplicated: {
                    nullable: true,
                    serializedName: "duplicated",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTagCreateEntry"
                            }
                        }
                    }
                },
                exceeded: {
                    nullable: true,
                    serializedName: "exceeded",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTagCreateEntry"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageRegionCreateEntry = {
        serializedName: "ImageRegionCreateEntry",
        type: {
            name: "Composite",
            className: "ImageRegionCreateEntry",
            modelProperties: {
                imageId: {
                    required: true,
                    nullable: false,
                    serializedName: "imageId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagId: {
                    required: true,
                    nullable: false,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                left: {
                    required: true,
                    nullable: false,
                    serializedName: "left",
                    type: {
                        name: "Number"
                    }
                },
                top: {
                    required: true,
                    nullable: false,
                    serializedName: "top",
                    type: {
                        name: "Number"
                    }
                },
                width: {
                    required: true,
                    nullable: false,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    required: true,
                    nullable: false,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var ImageRegionCreateBatch = {
        serializedName: "ImageRegionCreateBatch",
        type: {
            name: "Composite",
            className: "ImageRegionCreateBatch",
            modelProperties: {
                regions: {
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegionCreateEntry"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageRegionCreateResult = {
        serializedName: "ImageRegionCreateResult",
        type: {
            name: "Composite",
            className: "ImageRegionCreateResult",
            modelProperties: {
                imageId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "imageId",
                    type: {
                        name: "Uuid"
                    }
                },
                regionId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "regionId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagName: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "tagName",
                    type: {
                        name: "String"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                tagId: {
                    required: true,
                    nullable: false,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                left: {
                    required: true,
                    nullable: false,
                    serializedName: "left",
                    type: {
                        name: "Number"
                    }
                },
                top: {
                    required: true,
                    nullable: false,
                    serializedName: "top",
                    type: {
                        name: "Number"
                    }
                },
                width: {
                    required: true,
                    nullable: false,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    required: true,
                    nullable: false,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var ImageRegionCreateSummary = {
        serializedName: "ImageRegionCreateSummary",
        type: {
            name: "Composite",
            className: "ImageRegionCreateSummary",
            modelProperties: {
                created: {
                    nullable: true,
                    serializedName: "created",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegionCreateResult"
                            }
                        }
                    }
                },
                duplicated: {
                    nullable: true,
                    serializedName: "duplicated",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegionCreateEntry"
                            }
                        }
                    }
                },
                exceeded: {
                    nullable: true,
                    serializedName: "exceeded",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegionCreateEntry"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageTag = {
        serializedName: "ImageTag",
        type: {
            name: "Composite",
            className: "ImageTag",
            modelProperties: {
                tagId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagName: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "tagName",
                    type: {
                        name: "String"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                }
            }
        }
    };
    var ImageRegion = {
        serializedName: "ImageRegion",
        type: {
            name: "Composite",
            className: "ImageRegion",
            modelProperties: {
                regionId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "regionId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagName: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "tagName",
                    type: {
                        name: "String"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                tagId: {
                    required: true,
                    nullable: false,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                left: {
                    required: true,
                    nullable: false,
                    serializedName: "left",
                    type: {
                        name: "Number"
                    }
                },
                top: {
                    required: true,
                    nullable: false,
                    serializedName: "top",
                    type: {
                        name: "Number"
                    }
                },
                width: {
                    required: true,
                    nullable: false,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    required: true,
                    nullable: false,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var Image = {
        serializedName: "Image",
        type: {
            name: "Composite",
            className: "Image",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                width: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                },
                resizedImageUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "resizedImageUri",
                    type: {
                        name: "String"
                    }
                },
                thumbnailUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "thumbnailUri",
                    type: {
                        name: "String"
                    }
                },
                originalImageUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "originalImageUri",
                    type: {
                        name: "String"
                    }
                },
                tags: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "tags",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTag"
                            }
                        }
                    }
                },
                regions: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegion"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageCreateResult = {
        serializedName: "ImageCreateResult",
        type: {
            name: "Composite",
            className: "ImageCreateResult",
            modelProperties: {
                sourceUrl: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "sourceUrl",
                    type: {
                        name: "String"
                    }
                },
                status: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "status",
                    type: {
                        name: "String"
                    }
                },
                image: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "image",
                    type: {
                        name: "Composite",
                        className: "Image"
                    }
                }
            }
        }
    };
    var ImageCreateSummary = {
        serializedName: "ImageCreateSummary",
        type: {
            name: "Composite",
            className: "ImageCreateSummary",
            modelProperties: {
                isBatchSuccessful: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "isBatchSuccessful",
                    type: {
                        name: "Boolean"
                    }
                },
                images: {
                    readOnly: true,
                    serializedName: "images",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageCreateResult"
                            }
                        }
                    }
                }
            }
        }
    };
    var Region = {
        serializedName: "Region",
        type: {
            name: "Composite",
            className: "Region",
            modelProperties: {
                tagId: {
                    required: true,
                    nullable: false,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                left: {
                    required: true,
                    nullable: false,
                    serializedName: "left",
                    type: {
                        name: "Number"
                    }
                },
                top: {
                    required: true,
                    nullable: false,
                    serializedName: "top",
                    type: {
                        name: "Number"
                    }
                },
                width: {
                    required: true,
                    nullable: false,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    required: true,
                    nullable: false,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var ImageFileCreateEntry = {
        serializedName: "ImageFileCreateEntry",
        type: {
            name: "Composite",
            className: "ImageFileCreateEntry",
            modelProperties: {
                name: {
                    nullable: false,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                contents: {
                    nullable: false,
                    serializedName: "contents",
                    type: {
                        name: "ByteArray"
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                },
                regions: {
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Region"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageFileCreateBatch = {
        serializedName: "ImageFileCreateBatch",
        type: {
            name: "Composite",
            className: "ImageFileCreateBatch",
            modelProperties: {
                images: {
                    serializedName: "images",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageFileCreateEntry"
                            }
                        }
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageUrlCreateEntry = {
        serializedName: "ImageUrlCreateEntry",
        type: {
            name: "Composite",
            className: "ImageUrlCreateEntry",
            modelProperties: {
                url: {
                    required: true,
                    nullable: false,
                    serializedName: "url",
                    type: {
                        name: "String"
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                },
                regions: {
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Region"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageUrlCreateBatch = {
        serializedName: "ImageUrlCreateBatch",
        type: {
            name: "Composite",
            className: "ImageUrlCreateBatch",
            modelProperties: {
                images: {
                    serializedName: "images",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageUrlCreateEntry"
                            }
                        }
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageIdCreateEntry = {
        serializedName: "ImageIdCreateEntry",
        type: {
            name: "Composite",
            className: "ImageIdCreateEntry",
            modelProperties: {
                id: {
                    nullable: false,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                },
                regions: {
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Region"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageIdCreateBatch = {
        serializedName: "ImageIdCreateBatch",
        type: {
            name: "Composite",
            className: "ImageIdCreateBatch",
            modelProperties: {
                images: {
                    serializedName: "images",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageIdCreateEntry"
                            }
                        }
                    }
                },
                tagIds: {
                    serializedName: "tagIds",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Uuid"
                            }
                        }
                    }
                }
            }
        }
    };
    var BoundingBox = {
        serializedName: "BoundingBox",
        type: {
            name: "Composite",
            className: "BoundingBox",
            modelProperties: {
                left: {
                    required: true,
                    nullable: false,
                    serializedName: "left",
                    type: {
                        name: "Number"
                    }
                },
                top: {
                    required: true,
                    nullable: false,
                    serializedName: "top",
                    type: {
                        name: "Number"
                    }
                },
                width: {
                    required: true,
                    nullable: false,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    required: true,
                    nullable: false,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var RegionProposal = {
        serializedName: "RegionProposal",
        type: {
            name: "Composite",
            className: "RegionProposal",
            modelProperties: {
                confidence: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "confidence",
                    type: {
                        name: "Number"
                    }
                },
                boundingBox: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "boundingBox",
                    type: {
                        name: "Composite",
                        className: "BoundingBox"
                    }
                }
            }
        }
    };
    var ImageRegionProposal = {
        serializedName: "ImageRegionProposal",
        type: {
            name: "Composite",
            className: "ImageRegionProposal",
            modelProperties: {
                projectId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "projectId",
                    type: {
                        name: "Uuid"
                    }
                },
                imageId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "imageId",
                    type: {
                        name: "Uuid"
                    }
                },
                proposals: {
                    readOnly: true,
                    serializedName: "proposals",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "RegionProposal"
                            }
                        }
                    }
                }
            }
        }
    };
    var ImageUrl = {
        serializedName: "ImageUrl",
        type: {
            name: "Composite",
            className: "ImageUrl",
            modelProperties: {
                url: {
                    required: true,
                    nullable: false,
                    serializedName: "url",
                    type: {
                        name: "String"
                    }
                }
            }
        }
    };
    var Prediction = {
        serializedName: "Prediction",
        type: {
            name: "Composite",
            className: "Prediction",
            modelProperties: {
                probability: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "probability",
                    type: {
                        name: "Number"
                    }
                },
                tagId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "tagId",
                    type: {
                        name: "Uuid"
                    }
                },
                tagName: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "tagName",
                    type: {
                        name: "String"
                    }
                },
                boundingBox: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "boundingBox",
                    type: {
                        name: "Composite",
                        className: "BoundingBox"
                    }
                }
            }
        }
    };
    var ImagePrediction = {
        serializedName: "ImagePrediction",
        type: {
            name: "Composite",
            className: "ImagePrediction",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                project: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "project",
                    type: {
                        name: "Uuid"
                    }
                },
                iteration: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "iteration",
                    type: {
                        name: "Uuid"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                predictions: {
                    readOnly: true,
                    serializedName: "predictions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Prediction"
                            }
                        }
                    }
                }
            }
        }
    };
    var PredictionQueryTag = {
        serializedName: "PredictionQueryTag",
        type: {
            name: "Composite",
            className: "PredictionQueryTag",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                minThreshold: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "minThreshold",
                    type: {
                        name: "Number"
                    }
                },
                maxThreshold: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "maxThreshold",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var PredictionQueryToken = {
        serializedName: "PredictionQueryToken",
        type: {
            name: "Composite",
            className: "PredictionQueryToken",
            modelProperties: {
                session: {
                    nullable: false,
                    serializedName: "session",
                    type: {
                        name: "String"
                    }
                },
                continuation: {
                    nullable: false,
                    serializedName: "continuation",
                    type: {
                        name: "String"
                    }
                },
                maxCount: {
                    nullable: false,
                    serializedName: "maxCount",
                    type: {
                        name: "Number"
                    }
                },
                orderBy: {
                    nullable: false,
                    serializedName: "orderBy",
                    type: {
                        name: "String"
                    }
                },
                tags: {
                    serializedName: "tags",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "PredictionQueryTag"
                            }
                        }
                    }
                },
                iterationId: {
                    nullable: true,
                    serializedName: "iterationId",
                    type: {
                        name: "Uuid"
                    }
                },
                startTime: {
                    nullable: true,
                    serializedName: "startTime",
                    type: {
                        name: "DateTime"
                    }
                },
                endTime: {
                    nullable: true,
                    serializedName: "endTime",
                    type: {
                        name: "DateTime"
                    }
                },
                application: {
                    nullable: false,
                    serializedName: "application",
                    type: {
                        name: "String"
                    }
                }
            }
        }
    };
    var StoredImagePrediction = {
        serializedName: "StoredImagePrediction",
        type: {
            name: "Composite",
            className: "StoredImagePrediction",
            modelProperties: {
                resizedImageUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "resizedImageUri",
                    type: {
                        name: "String"
                    }
                },
                thumbnailUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "thumbnailUri",
                    type: {
                        name: "String"
                    }
                },
                originalImageUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "originalImageUri",
                    type: {
                        name: "String"
                    }
                },
                domain: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "domain",
                    type: {
                        name: "Uuid"
                    }
                },
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                project: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "project",
                    type: {
                        name: "Uuid"
                    }
                },
                iteration: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "iteration",
                    type: {
                        name: "Uuid"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                predictions: {
                    readOnly: true,
                    serializedName: "predictions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Prediction"
                            }
                        }
                    }
                }
            }
        }
    };
    var PredictionQueryResult = {
        serializedName: "PredictionQueryResult",
        type: {
            name: "Composite",
            className: "PredictionQueryResult",
            modelProperties: {
                token: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "token",
                    type: {
                        name: "Composite",
                        className: "PredictionQueryToken"
                    }
                },
                results: {
                    readOnly: true,
                    serializedName: "results",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "StoredImagePrediction"
                            }
                        }
                    }
                }
            }
        }
    };
    var TagPerformance = {
        serializedName: "TagPerformance",
        type: {
            name: "Composite",
            className: "TagPerformance",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                name: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                precision: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "precision",
                    type: {
                        name: "Number"
                    }
                },
                precisionStdDeviation: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "precisionStdDeviation",
                    type: {
                        name: "Number"
                    }
                },
                recall: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "recall",
                    type: {
                        name: "Number"
                    }
                },
                recallStdDeviation: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "recallStdDeviation",
                    type: {
                        name: "Number"
                    }
                },
                averagePrecision: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "averagePrecision",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var IterationPerformance = {
        serializedName: "IterationPerformance",
        type: {
            name: "Composite",
            className: "IterationPerformance",
            modelProperties: {
                perTagPerformance: {
                    readOnly: true,
                    serializedName: "perTagPerformance",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "TagPerformance"
                            }
                        }
                    }
                },
                precision: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "precision",
                    type: {
                        name: "Number"
                    }
                },
                precisionStdDeviation: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "precisionStdDeviation",
                    type: {
                        name: "Number"
                    }
                },
                recall: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "recall",
                    type: {
                        name: "Number"
                    }
                },
                recallStdDeviation: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "recallStdDeviation",
                    type: {
                        name: "Number"
                    }
                },
                averagePrecision: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "averagePrecision",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var ImagePerformance = {
        serializedName: "ImagePerformance",
        type: {
            name: "Composite",
            className: "ImagePerformance",
            modelProperties: {
                predictions: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "predictions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Prediction"
                            }
                        }
                    }
                },
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                width: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "width",
                    type: {
                        name: "Number"
                    }
                },
                height: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "height",
                    type: {
                        name: "Number"
                    }
                },
                imageUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "imageUri",
                    type: {
                        name: "String"
                    }
                },
                thumbnailUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "thumbnailUri",
                    type: {
                        name: "String"
                    }
                },
                tags: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "tags",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageTag"
                            }
                        }
                    }
                },
                regions: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "regions",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImageRegion"
                            }
                        }
                    }
                }
            }
        }
    };
    var ProjectSettings = {
        serializedName: "ProjectSettings",
        type: {
            name: "Composite",
            className: "ProjectSettings",
            modelProperties: {
                domainId: {
                    nullable: false,
                    serializedName: "domainId",
                    type: {
                        name: "Uuid"
                    }
                },
                classificationType: {
                    nullable: true,
                    serializedName: "classificationType",
                    type: {
                        name: "String"
                    }
                },
                targetExportPlatforms: {
                    serializedName: "targetExportPlatforms",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "String"
                            }
                        }
                    }
                }
            }
        }
    };
    var Project = {
        serializedName: "Project",
        type: {
            name: "Composite",
            className: "Project",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                name: {
                    required: true,
                    nullable: false,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                description: {
                    required: true,
                    nullable: false,
                    serializedName: "description",
                    type: {
                        name: "String"
                    }
                },
                settings: {
                    required: true,
                    nullable: false,
                    serializedName: "settings",
                    type: {
                        name: "Composite",
                        className: "ProjectSettings"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                lastModified: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "lastModified",
                    type: {
                        name: "DateTime"
                    }
                },
                thumbnailUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "thumbnailUri",
                    type: {
                        name: "String"
                    }
                },
                drModeEnabled: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "drModeEnabled",
                    type: {
                        name: "Boolean"
                    }
                }
            }
        }
    };
    var Iteration = {
        serializedName: "Iteration",
        type: {
            name: "Composite",
            className: "Iteration",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                name: {
                    required: true,
                    nullable: false,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                status: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "status",
                    type: {
                        name: "String"
                    }
                },
                created: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "created",
                    type: {
                        name: "DateTime"
                    }
                },
                lastModified: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "lastModified",
                    type: {
                        name: "DateTime"
                    }
                },
                trainedAt: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "trainedAt",
                    type: {
                        name: "DateTime"
                    }
                },
                projectId: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "projectId",
                    type: {
                        name: "Uuid"
                    }
                },
                exportable: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "exportable",
                    type: {
                        name: "Boolean"
                    }
                },
                exportableTo: {
                    readOnly: true,
                    serializedName: "exportableTo",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "String"
                            }
                        }
                    }
                },
                domainId: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "domainId",
                    type: {
                        name: "Uuid"
                    }
                },
                classificationType: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "classificationType",
                    type: {
                        name: "String"
                    }
                },
                trainingType: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "trainingType",
                    type: {
                        name: "String"
                    }
                },
                reservedBudgetInHours: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "reservedBudgetInHours",
                    type: {
                        name: "Number"
                    }
                },
                publishName: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "publishName",
                    type: {
                        name: "String"
                    }
                },
                originalPublishResourceId: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "originalPublishResourceId",
                    type: {
                        name: "String"
                    }
                }
            }
        }
    };
    var ExportModel = {
        serializedName: "Export",
        type: {
            name: "Composite",
            className: "ExportModel",
            modelProperties: {
                platform: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "platform",
                    type: {
                        name: "String"
                    }
                },
                status: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "status",
                    type: {
                        name: "String"
                    }
                },
                downloadUri: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "downloadUri",
                    type: {
                        name: "String"
                    }
                },
                flavor: {
                    nullable: true,
                    readOnly: true,
                    serializedName: "flavor",
                    type: {
                        name: "String"
                    }
                },
                newerVersionAvailable: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "newerVersionAvailable",
                    type: {
                        name: "Boolean"
                    }
                }
            }
        }
    };
    var Tag = {
        serializedName: "Tag",
        type: {
            name: "Composite",
            className: "Tag",
            modelProperties: {
                id: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "id",
                    type: {
                        name: "Uuid"
                    }
                },
                name: {
                    required: true,
                    nullable: false,
                    serializedName: "name",
                    type: {
                        name: "String"
                    }
                },
                description: {
                    required: true,
                    nullable: false,
                    serializedName: "description",
                    type: {
                        name: "String"
                    }
                },
                type: {
                    required: true,
                    nullable: false,
                    serializedName: "type",
                    type: {
                        name: "String"
                    }
                },
                imageCount: {
                    nullable: false,
                    readOnly: true,
                    serializedName: "imageCount",
                    type: {
                        name: "Number"
                    }
                }
            }
        }
    };
    var CustomVisionError = {
        serializedName: "CustomVisionError",
        type: {
            name: "Composite",
            className: "CustomVisionError",
            modelProperties: {
                code: {
                    required: true,
                    serializedName: "code",
                    type: {
                        name: "String"
                    }
                },
                message: {
                    required: true,
                    serializedName: "message",
                    type: {
                        name: "String"
                    }
                }
            }
        }
    };

    var Mappers = /*#__PURE__*/Object.freeze({
        Domain: Domain,
        ImageTagCreateEntry: ImageTagCreateEntry,
        ImageTagCreateBatch: ImageTagCreateBatch,
        ImageTagCreateSummary: ImageTagCreateSummary,
        ImageRegionCreateEntry: ImageRegionCreateEntry,
        ImageRegionCreateBatch: ImageRegionCreateBatch,
        ImageRegionCreateResult: ImageRegionCreateResult,
        ImageRegionCreateSummary: ImageRegionCreateSummary,
        ImageTag: ImageTag,
        ImageRegion: ImageRegion,
        Image: Image,
        ImageCreateResult: ImageCreateResult,
        ImageCreateSummary: ImageCreateSummary,
        Region: Region,
        ImageFileCreateEntry: ImageFileCreateEntry,
        ImageFileCreateBatch: ImageFileCreateBatch,
        ImageUrlCreateEntry: ImageUrlCreateEntry,
        ImageUrlCreateBatch: ImageUrlCreateBatch,
        ImageIdCreateEntry: ImageIdCreateEntry,
        ImageIdCreateBatch: ImageIdCreateBatch,
        BoundingBox: BoundingBox,
        RegionProposal: RegionProposal,
        ImageRegionProposal: ImageRegionProposal,
        ImageUrl: ImageUrl,
        Prediction: Prediction,
        ImagePrediction: ImagePrediction,
        PredictionQueryTag: PredictionQueryTag,
        PredictionQueryToken: PredictionQueryToken,
        StoredImagePrediction: StoredImagePrediction,
        PredictionQueryResult: PredictionQueryResult,
        TagPerformance: TagPerformance,
        IterationPerformance: IterationPerformance,
        ImagePerformance: ImagePerformance,
        ProjectSettings: ProjectSettings,
        Project: Project,
        Iteration: Iteration,
        ExportModel: ExportModel,
        Tag: Tag,
        CustomVisionError: CustomVisionError
    });

    /*
     * Copyright (c) Microsoft Corporation. All rights reserved.
     * Licensed under the MIT License. See License.txt in the project root for
     * license information.
     *
     * Code generated by Microsoft (R) AutoRest Code Generator.
     * Changes may cause incorrect behavior and will be lost if the code is
     * regenerated.
     */
    var apiKey = {
        parameterPath: "apiKey",
        mapper: {
            required: true,
            serializedName: "Training-Key",
            type: {
                name: "String"
            }
        }
    };
    var classificationType = {
        parameterPath: [
            "options",
            "classificationType"
        ],
        mapper: {
            serializedName: "classificationType",
            type: {
                name: "String"
            }
        }
    };
    var description = {
        parameterPath: [
            "options",
            "description"
        ],
        mapper: {
            serializedName: "description",
            type: {
                name: "String"
            }
        }
    };
    var domainId0 = {
        parameterPath: "domainId",
        mapper: {
            required: true,
            serializedName: "domainId",
            type: {
                name: "Uuid"
            }
        }
    };
    var domainId1 = {
        parameterPath: [
            "options",
            "domainId"
        ],
        mapper: {
            serializedName: "domainId",
            type: {
                name: "Uuid"
            }
        }
    };
    var endpoint = {
        parameterPath: "endpoint",
        mapper: {
            required: true,
            serializedName: "Endpoint",
            defaultValue: '',
            type: {
                name: "String"
            }
        },
        skipEncoding: true
    };
    var flavor = {
        parameterPath: [
            "options",
            "flavor"
        ],
        mapper: {
            serializedName: "flavor",
            type: {
                name: "String"
            }
        }
    };
    var forceTrain = {
        parameterPath: [
            "options",
            "forceTrain"
        ],
        mapper: {
            serializedName: "forceTrain",
            defaultValue: false,
            type: {
                name: "Boolean"
            }
        }
    };
    var ids = {
        parameterPath: "ids",
        mapper: {
            required: true,
            serializedName: "ids",
            constraints: {
                MaxItems: 64,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var imageData = {
        parameterPath: "imageData",
        mapper: {
            required: true,
            serializedName: "imageData",
            type: {
                name: "Stream"
            }
        }
    };
    var imageId = {
        parameterPath: "imageId",
        mapper: {
            required: true,
            serializedName: "imageId",
            type: {
                name: "Uuid"
            }
        }
    };
    var imageIds0 = {
        parameterPath: "imageIds",
        mapper: {
            required: true,
            serializedName: "imageIds",
            constraints: {
                MaxItems: 64,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var imageIds1 = {
        parameterPath: [
            "options",
            "imageIds"
        ],
        mapper: {
            serializedName: "imageIds",
            constraints: {
                MaxItems: 256,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var imageIds2 = {
        parameterPath: "imageIds",
        mapper: {
            required: true,
            serializedName: "imageIds",
            constraints: {
                MaxItems: 256,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var iterationId0 = {
        parameterPath: [
            "options",
            "iterationId"
        ],
        mapper: {
            serializedName: "iterationId",
            type: {
                name: "Uuid"
            }
        }
    };
    var iterationId1 = {
        parameterPath: "iterationId",
        mapper: {
            required: true,
            serializedName: "iterationId",
            type: {
                name: "Uuid"
            }
        }
    };
    var name = {
        parameterPath: "name",
        mapper: {
            required: true,
            serializedName: "name",
            type: {
                name: "String"
            }
        }
    };
    var notificationEmailAddress = {
        parameterPath: [
            "options",
            "notificationEmailAddress"
        ],
        mapper: {
            serializedName: "notificationEmailAddress",
            type: {
                name: "String"
            }
        }
    };
    var orderBy = {
        parameterPath: [
            "options",
            "orderBy"
        ],
        mapper: {
            serializedName: "orderBy",
            type: {
                name: "String"
            }
        }
    };
    var overlapThreshold = {
        parameterPath: [
            "options",
            "overlapThreshold"
        ],
        mapper: {
            serializedName: "overlapThreshold",
            type: {
                name: "Number"
            }
        }
    };
    var platform = {
        parameterPath: "platform",
        mapper: {
            required: true,
            serializedName: "platform",
            type: {
                name: "String"
            }
        }
    };
    var predictionId = {
        parameterPath: "predictionId",
        mapper: {
            required: true,
            serializedName: "predictionId",
            type: {
                name: "String"
            }
        }
    };
    var projectId = {
        parameterPath: "projectId",
        mapper: {
            required: true,
            serializedName: "projectId",
            type: {
                name: "Uuid"
            }
        }
    };
    var publishName = {
        parameterPath: "publishName",
        mapper: {
            required: true,
            serializedName: "publishName",
            type: {
                name: "String"
            }
        }
    };
    var regionIds = {
        parameterPath: "regionIds",
        mapper: {
            required: true,
            serializedName: "regionIds",
            constraints: {
                MaxItems: 64,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var reservedBudgetInHours = {
        parameterPath: [
            "options",
            "reservedBudgetInHours"
        ],
        mapper: {
            serializedName: "reservedBudgetInHours",
            defaultValue: 0,
            type: {
                name: "Number"
            }
        }
    };
    var skip = {
        parameterPath: [
            "options",
            "skip"
        ],
        mapper: {
            serializedName: "skip",
            defaultValue: 0,
            type: {
                name: "Number"
            }
        }
    };
    var tagId = {
        parameterPath: "tagId",
        mapper: {
            required: true,
            serializedName: "tagId",
            type: {
                name: "Uuid"
            }
        }
    };
    var tagIds0 = {
        parameterPath: [
            "options",
            "tagIds"
        ],
        mapper: {
            serializedName: "tagIds",
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var tagIds1 = {
        parameterPath: "tagIds",
        mapper: {
            required: true,
            serializedName: "tagIds",
            constraints: {
                MaxItems: 20,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var tagIds2 = {
        parameterPath: [
            "options",
            "tagIds"
        ],
        mapper: {
            serializedName: "tagIds",
            constraints: {
                MaxItems: 20,
                MinItems: 0
            },
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Uuid"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var take = {
        parameterPath: [
            "options",
            "take"
        ],
        mapper: {
            serializedName: "take",
            defaultValue: 50,
            constraints: {
                InclusiveMaximum: 256,
                InclusiveMinimum: 0
            },
            type: {
                name: "Number"
            }
        }
    };
    var targetExportPlatforms = {
        parameterPath: [
            "options",
            "targetExportPlatforms"
        ],
        mapper: {
            serializedName: "targetExportPlatforms",
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "String"
                    }
                }
            }
        },
        collectionFormat: msRest.QueryCollectionFormat.Csv
    };
    var threshold = {
        parameterPath: [
            "options",
            "threshold"
        ],
        mapper: {
            serializedName: "threshold",
            type: {
                name: "Number"
            }
        }
    };
    var trainingType = {
        parameterPath: [
            "options",
            "trainingType"
        ],
        mapper: {
            serializedName: "trainingType",
            type: {
                name: "String"
            }
        }
    };
    var type = {
        parameterPath: [
            "options",
            "type"
        ],
        mapper: {
            serializedName: "type",
            type: {
                name: "String"
            }
        }
    };

    /*
     * Copyright (c) Microsoft Corporation. All rights reserved.
     * Licensed under the MIT License. See License.txt in the project root for
     * license information.
     *
     * Code generated by Microsoft (R) AutoRest Code Generator.
     * Changes may cause incorrect behavior and will be lost if the code is
     * regenerated.
     */
    var packageName = "@azure/cognitiveservices-customvision-training";
    var packageVersion = "4.0.0";
    var TrainingAPIClientContext = /** @class */ (function (_super) {
        __extends(TrainingAPIClientContext, _super);
        /**
         * Initializes a new instance of the TrainingAPIClientContext class.
         * @param apiKey API key.
         * @param endpoint Supported Cognitive Services endpoints.
         * @param [options] The parameter options
         */
        function TrainingAPIClientContext(apiKey, endpoint, options) {
            var _this = this;
            if (apiKey == undefined) {
                throw new Error("'apiKey' cannot be null.");
            }
            if (endpoint == undefined) {
                throw new Error("'endpoint' cannot be null.");
            }
            if (!options) {
                options = {};
            }
            if (!options.userAgent) {
                var defaultUserAgent = msRest.getDefaultUserAgentValue();
                options.userAgent = packageName + "/" + packageVersion + " " + defaultUserAgent;
            }
            _this = _super.call(this, undefined, options) || this;
            _this.baseUri = "{Endpoint}/customvision/v3.0/training";
            _this.requestContentType = "application/json; charset=utf-8";
            _this.apiKey = apiKey;
            _this.endpoint = endpoint;
            return _this;
        }
        return TrainingAPIClientContext;
    }(msRest.ServiceClient));

    /*
     * Copyright (c) Microsoft Corporation. All rights reserved.
     * Licensed under the MIT License. See License.txt in the project root for
     * license information.
     *
     * Code generated by Microsoft (R) AutoRest Code Generator.
     * Changes may cause incorrect behavior and will be lost if the code is
     * regenerated.
     */
    var TrainingAPIClient = /** @class */ (function (_super) {
        __extends(TrainingAPIClient, _super);
        /**
         * Initializes a new instance of the TrainingAPIClient class.
         * @param apiKey API key.
         * @param endpoint Supported Cognitive Services endpoints.
         * @param [options] The parameter options
         */
        function TrainingAPIClient(apiKey, endpoint, options) {
            return _super.call(this, apiKey, endpoint, options) || this;
        }
        TrainingAPIClient.prototype.getDomains = function (options, callback) {
            return this.sendOperationRequest({
                options: options
            }, getDomainsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getDomain = function (domainId, options, callback) {
            return this.sendOperationRequest({
                domainId: domainId,
                options: options
            }, getDomainOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getTaggedImageCount = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getTaggedImageCountOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getUntaggedImageCount = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getUntaggedImageCountOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImageTags = function (projectId, batch, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                batch: batch,
                options: options
            }, createImageTagsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteImageTags = function (projectId, imageIds, tagIds, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageIds: imageIds,
                tagIds: tagIds,
                options: options
            }, deleteImageTagsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImageRegions = function (projectId, batch, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                batch: batch,
                options: options
            }, createImageRegionsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteImageRegions = function (projectId, regionIds, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                regionIds: regionIds,
                options: options
            }, deleteImageRegionsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getTaggedImages = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getTaggedImagesOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getUntaggedImages = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getUntaggedImagesOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getImagesByIds = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getImagesByIdsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImagesFromData = function (projectId, imageData, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageData: imageData,
                options: options
            }, createImagesFromDataOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteImages = function (projectId, imageIds, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageIds: imageIds,
                options: options
            }, deleteImagesOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImagesFromFiles = function (projectId, batch, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                batch: batch,
                options: options
            }, createImagesFromFilesOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImagesFromUrls = function (projectId, batch, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                batch: batch,
                options: options
            }, createImagesFromUrlsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createImagesFromPredictions = function (projectId, batch, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                batch: batch,
                options: options
            }, createImagesFromPredictionsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getImageRegionProposals = function (projectId, imageId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageId: imageId,
                options: options
            }, getImageRegionProposalsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deletePrediction = function (projectId, ids, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                ids: ids,
                options: options
            }, deletePredictionOperationSpec, callback);
        };
        TrainingAPIClient.prototype.quickTestImageUrl = function (projectId, imageUrl, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageUrl: imageUrl,
                options: options
            }, quickTestImageUrlOperationSpec, callback);
        };
        TrainingAPIClient.prototype.quickTestImage = function (projectId, imageData, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                imageData: imageData,
                options: options
            }, quickTestImageOperationSpec, callback);
        };
        TrainingAPIClient.prototype.queryPredictions = function (projectId, query, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                query: query,
                options: options
            }, queryPredictionsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getIterationPerformance = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, getIterationPerformanceOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getImagePerformances = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, getImagePerformancesOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getImagePerformanceCount = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, getImagePerformanceCountOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getProjects = function (options, callback) {
            return this.sendOperationRequest({
                options: options
            }, getProjectsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createProject = function (name, options, callback) {
            return this.sendOperationRequest({
                name: name,
                options: options
            }, createProjectOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getProject = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getProjectOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteProject = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, deleteProjectOperationSpec, callback);
        };
        TrainingAPIClient.prototype.updateProject = function (projectId, updatedProject, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                updatedProject: updatedProject,
                options: options
            }, updateProjectOperationSpec, callback);
        };
        TrainingAPIClient.prototype.trainProject = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, trainProjectOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getIterations = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getIterationsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getIteration = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, getIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteIteration = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, deleteIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.updateIteration = function (projectId, iterationId, updatedIteration, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                updatedIteration: updatedIteration,
                options: options
            }, updateIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.publishIteration = function (projectId, iterationId, publishName, predictionId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                publishName: publishName,
                predictionId: predictionId,
                options: options
            }, publishIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.unpublishIteration = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, unpublishIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getExports = function (projectId, iterationId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                options: options
            }, getExportsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.exportIteration = function (projectId, iterationId, platform, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                iterationId: iterationId,
                platform: platform,
                options: options
            }, exportIterationOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getTag = function (projectId, tagId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                tagId: tagId,
                options: options
            }, getTagOperationSpec, callback);
        };
        TrainingAPIClient.prototype.deleteTag = function (projectId, tagId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                tagId: tagId,
                options: options
            }, deleteTagOperationSpec, callback);
        };
        TrainingAPIClient.prototype.updateTag = function (projectId, tagId, updatedTag, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                tagId: tagId,
                updatedTag: updatedTag,
                options: options
            }, updateTagOperationSpec, callback);
        };
        TrainingAPIClient.prototype.getTags = function (projectId, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                options: options
            }, getTagsOperationSpec, callback);
        };
        TrainingAPIClient.prototype.createTag = function (projectId, name, options, callback) {
            return this.sendOperationRequest({
                projectId: projectId,
                name: name,
                options: options
            }, createTagOperationSpec, callback);
        };
        return TrainingAPIClient;
    }(TrainingAPIClientContext));
    // Operation Specifications
    var serializer = new msRest.Serializer(Mappers);
    var getDomainsOperationSpec = {
        httpMethod: "GET",
        path: "domains",
        urlParameters: [
            endpoint
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Domain"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getDomainOperationSpec = {
        httpMethod: "GET",
        path: "domains/{domainId}",
        urlParameters: [
            endpoint,
            domainId0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Domain
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getTaggedImageCountOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/images/tagged/count",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0,
            tagIds0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Number"
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getUntaggedImageCountOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/images/untagged/count",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Number"
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImageTagsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/tags",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "batch",
            mapper: __assign(__assign({}, ImageTagCreateBatch), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImageTagCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteImageTagsOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/images/tags",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            imageIds0,
            tagIds1
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImageRegionsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/regions",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "batch",
            mapper: __assign(__assign({}, ImageRegionCreateBatch), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImageRegionCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteImageRegionsOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/images/regions",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            regionIds
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getTaggedImagesOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/images/tagged",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0,
            tagIds2,
            orderBy,
            take,
            skip
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Image"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getUntaggedImagesOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/images/untagged",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0,
            orderBy,
            take,
            skip
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Image"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getImagesByIdsOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/images/id",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            imageIds1,
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Image"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImagesFromDataOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            tagIds2
        ],
        headerParameters: [
            apiKey
        ],
        formDataParameters: [
            imageData
        ],
        contentType: "multipart/form-data",
        responses: {
            200: {
                bodyMapper: ImageCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteImagesOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/images",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            imageIds2
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImagesFromFilesOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/files",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "batch",
            mapper: __assign(__assign({}, ImageFileCreateBatch), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImageCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImagesFromUrlsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/urls",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "batch",
            mapper: __assign(__assign({}, ImageUrlCreateBatch), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImageCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createImagesFromPredictionsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/predictions",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "batch",
            mapper: __assign(__assign({}, ImageIdCreateBatch), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImageCreateSummary
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getImageRegionProposalsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/images/{imageId}/regionproposals",
        urlParameters: [
            endpoint,
            projectId,
            imageId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: ImageRegionProposal
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deletePredictionOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/predictions",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            ids
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var quickTestImageUrlOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/quicktest/url",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "imageUrl",
            mapper: __assign(__assign({}, ImageUrl), { required: true })
        },
        responses: {
            200: {
                bodyMapper: ImagePrediction
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var quickTestImageOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/quicktest/image",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        formDataParameters: [
            imageData
        ],
        contentType: "multipart/form-data",
        responses: {
            200: {
                bodyMapper: ImagePrediction
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var queryPredictionsOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/predictions/query",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "query",
            mapper: __assign(__assign({}, PredictionQueryToken), { required: true })
        },
        responses: {
            200: {
                bodyMapper: PredictionQueryResult
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getIterationPerformanceOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations/{iterationId}/performance",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        queryParameters: [
            threshold,
            overlapThreshold
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: IterationPerformance
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getImagePerformancesOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations/{iterationId}/performance/images",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        queryParameters: [
            tagIds2,
            orderBy,
            take,
            skip
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ImagePerformance"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getImagePerformanceCountOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations/{iterationId}/performance/images/count",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        queryParameters: [
            tagIds0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Number"
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getProjectsOperationSpec = {
        httpMethod: "GET",
        path: "projects",
        urlParameters: [
            endpoint
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Project"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createProjectOperationSpec = {
        httpMethod: "POST",
        path: "projects",
        urlParameters: [
            endpoint
        ],
        queryParameters: [
            name,
            description,
            domainId1,
            classificationType,
            targetExportPlatforms
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Project
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getProjectOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Project
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteProjectOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var updateProjectOperationSpec = {
        httpMethod: "PATCH",
        path: "projects/{projectId}",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "updatedProject",
            mapper: __assign(__assign({}, Project), { required: true })
        },
        responses: {
            200: {
                bodyMapper: Project
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var trainProjectOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/train",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            trainingType,
            reservedBudgetInHours,
            forceTrain,
            notificationEmailAddress
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Iteration
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getIterationsOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations",
        urlParameters: [
            endpoint,
            projectId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Iteration"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getIterationOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations/{iterationId}",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Iteration
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteIterationOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/iterations/{iterationId}",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var updateIterationOperationSpec = {
        httpMethod: "PATCH",
        path: "projects/{projectId}/iterations/{iterationId}",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "updatedIteration",
            mapper: __assign(__assign({}, Iteration), { required: true })
        },
        responses: {
            200: {
                bodyMapper: Iteration
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var publishIterationOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/iterations/{iterationId}/publish",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        queryParameters: [
            publishName,
            predictionId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Boolean"
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var unpublishIterationOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/iterations/{iterationId}/publish",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getExportsOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/iterations/{iterationId}/export",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "ExportModel"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var exportIterationOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/iterations/{iterationId}/export",
        urlParameters: [
            endpoint,
            projectId,
            iterationId1
        ],
        queryParameters: [
            platform,
            flavor
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: ExportModel
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getTagOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/tags/{tagId}",
        urlParameters: [
            endpoint,
            projectId,
            tagId
        ],
        queryParameters: [
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Tag
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var deleteTagOperationSpec = {
        httpMethod: "DELETE",
        path: "projects/{projectId}/tags/{tagId}",
        urlParameters: [
            endpoint,
            projectId,
            tagId
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            204: {},
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var updateTagOperationSpec = {
        httpMethod: "PATCH",
        path: "projects/{projectId}/tags/{tagId}",
        urlParameters: [
            endpoint,
            projectId,
            tagId
        ],
        headerParameters: [
            apiKey
        ],
        requestBody: {
            parameterPath: "updatedTag",
            mapper: __assign(__assign({}, Tag), { required: true })
        },
        responses: {
            200: {
                bodyMapper: Tag
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var getTagsOperationSpec = {
        httpMethod: "GET",
        path: "projects/{projectId}/tags",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            iterationId0
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: {
                    serializedName: "parsedResponse",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "Tag"
                            }
                        }
                    }
                }
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };
    var createTagOperationSpec = {
        httpMethod: "POST",
        path: "projects/{projectId}/tags",
        urlParameters: [
            endpoint,
            projectId
        ],
        queryParameters: [
            name,
            description,
            type
        ],
        headerParameters: [
            apiKey
        ],
        responses: {
            200: {
                bodyMapper: Tag
            },
            default: {
                bodyMapper: CustomVisionError
            }
        },
        serializer: serializer
    };

    exports.TrainingAPIClient = TrainingAPIClient;
    exports.TrainingAPIClientContext = TrainingAPIClientContext;
    exports.TrainingAPIMappers = Mappers;
    exports.TrainingAPIModels = index;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
